website/docs/language/checks/index.mdx

```diff
@@ -0,0 +1,83 @@
+---
+page_title: Checks - Configuration Language
+description: >-
+  Check custom requirements for infrastructure to provide ongoing and continuous validation.
+---
+
+# Checks
+
+-> **Note:** Check blocks are only available in Terraform v1.5.0 and later.
+
+You can use `check` blocks to validate your infrastructure outside the usual resource lifecycle. Check blocks address a gap between post-apply and functional validation of infrastructure. Check blocks allow you to define [Custom Conditions](/terraform/language/expressions/custom-conditions) that execute on every Terraform plan or apply operation without affecting the overall status of the operation. They execute as the last step of a plan/apply, after the infrastructure has been planned/provisioned.
+
+## Checks Syntax
+
+A `check` block declares a check with a given local name, zero to one scoped [data sources](#scoped-data-sources), and one to many [assertions](#assertions).
+
+The following example loads the Terraform website and validates it is returning the expected status code of 200.
+
+```hcl
+check "health_check" {
+  data "http" "terraform_io" {
+    url = "https://www.terraform.io"
+  }
+
+  assert {
+    condition = data.http.terraform_io.status_code == 200
+    error_message = "${data.http.terraform_io.url} returned an unhealthy status code"
+  }
+}
+```
+
+### Scoped Data Sources
+
+A `check` block can optionally have a nested data source. This block behaves in the same way as normal/external [data sources](/terraform/language/data-sources), except it cannot be referenced from outside the enclosing check block. In addition, any errors or failures raised by the provider of the data source will be masked as warnings within the output and will not prevent the continued execution of the overall Terraform operation.
+
+A scoped data source allows you to validate the status of a piece of infrastructure outside the usual Terraform resource lifecycle. In the example in the previous section, if the `terraform_io` data source fails to load we would simply receive a warning that the `http` block couldn't be loaded rather than the blocking error that would be raised if the data source was defined outside a check block.
+
+Any data source from any provider can be used within a check block.
```

Comment:
I moved this first, and changed it to: "You can use any data source from any provider as a scoped data source within a `check` block."

Since it's now the first line, highlighting the scoped data source seemed important.